frontend/src/tw-components/Buttons.tsx

```diff
@@ -1,100 +1,100 @@
   "small-long": Typography.Title7,
   medium: Typography.Title6,
   "medium-long": Typography.Title6,
   large: Typography.Title5,
   "large-long": Typography.Title5,
-  "icon-only": Typography.Title7, // Not sure about this one
+  "icon-only": null,
 };
 
+// Shared base styles for all buttons
+const baseButtonStyles =
+  "transition-all duration-200 flex items-center justify-center bg-blue-dark text-white hover:bg-blue-dark-hover focus:bg-blue-dark-focused focus:outline-none active:bg-blue-dark-focused";
+
+const enabledDarkModeStyles =
+  "dark:bg-white dark:text-blue-dark dark:hover:bg-grey-light dark:focus:bg-[#D9DBDF] dark:active:bg-[#D9DBDF]";
+
+const disabledStyles = "pointer-events-none bg-grey text-grey-light";
+
+// Button component (handles both regular and icon buttons)
 const Button: React.FC<ButtonProps> = ({
   size = "medium",
-  state = "enabled",
+  disabled = false,
   children,
   onClick,
+  isIconButton = false,
 }) => {
   const TextComponent = buttonTypography[size];
 
   return (
     <button
-      className={`rounded-[64px] ${buttonSizes[size]} ${buttonColors[state]} transition-all duration-200`}
-      disabled={state === "disabled"}
+      className={clsx(
+        baseButtonStyles,
+        buttonSizes[size],
+        isIconButton ? "rounded-[24px]" : "rounded-[64px]",
+        disabled ? disabledStyles : enabledDarkModeStyles,
+      )}
+      disabled={disabled}
       onClick={onClick}
     >
-      <TextComponent>{children}</TextComponent>
+      {isIconButton ? (
+        <svg
+          xmlns="http://www.w3.org/2000/svg"
+          width="24"
+          height="24"
+          viewBox="0 0 24 24"
+        >
+          <path
+            fill="none"
+            stroke="currentColor"
+            strokeLinecap="round"
+            strokeLinejoin="round"
+            strokeWidth="2"
+            d="m20 20l-4.05-4.05m0 0a7 7 0 1 0-9.9-9.9a7 7 0 0 0 9.9 9.9"
+          />
+        </svg>
+      ) : (
+        TextComponent && <TextComponent>{children}</TextComponent>
+      )}
     </button>
   );
 };
 
-const IconButton: React.FC<IconButtonProps> = ({
-  state = "enabled",
-  onClick,
-}) => {
-  return (
-    <button
-      className={`rounded-[24px] ${buttonSizes["icon-only"]} ${buttonColors[state]} transition-all duration-200`}
-      disabled={state === "disabled"}
-      onClick={onClick}
-    >
-      <svg
-        xmlns="http://www.w3.org/2000/svg"
-        width="24"
-        height="24"
-        viewBox="0 0 24 24"
-      >
-        <path
-          fill="none"
-          stroke="currentColor"
-          strokeLinecap="round"
-          strokeLinejoin="round"
-          strokeWidth="2"
-          d="m20 20l-4.05-4.05m0 0a7 7 0 1 0-9.9-9.9a7 7 0 0 0 9.9 9.9"
-        />
-      </svg>
-    </button>
-  );
-};
+// Export separate IconButton
+const IconButton: React.FC<Omit<ButtonProps, "size" | "children">> = (
+  props,
+) => <Button {...props} size="icon-only" isIconButton />;
```

Comment:
```suggestion
) => {
  return (
    <Button {...props} size="icon-only" className="rounded-[24px]">
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="24"
        height="24"
        viewBox="0 0 24 24"
      >
        <path
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="m20 20l-4.05-4.05m0 0a7 7 0 1 0-9.9-9.9a7 7 0 0 0 9.9 9.9"
        />
      </svg>
    </Button>
  );
};
```

Ideally `<IconButton>` should extend the `<Button>` component, rather than `Button` having a flag to check if its an Icon button or not. 

Otherwise why not just use `<Button isIconButton size="icon-only">` instead of `<IconButton>` in the first place?

With this, we can simplify the Button component by removing the `isIconButton` property.